src/harness/test_harness_objects.py

```diff
@@ -0,0 +1,327 @@
+    heartbeat_request = {
+        'grantId': self.grant_id,
+        'cbsdId': self.grant_request['cbsdId'],
+        'operationState': 'GRANTED'
+      }
+    return heartbeat_request
+
+  def isGrantAuthorizedInLastHeartbeat(self):
+    return self.authorized_in_last_heartbeat
+
+  def isActive(self):
+    return not self.is_terminated
+
+
+class Cbsd(object):
+  """Holds the CBSD related parameters."""
+
+  def __init__(self, cbsd_id, registration_request, grant_ids, grant_requests):
+    """Constructor to store the registration request parameters.
+
+    Args:
+      cbsd_id: cbsd id of the device extracted from the registration response.
+      registration_request:A registrationRequest object as defined in the SAS-CBSD TS.
+      grant_ids: list of grant ids.
+      grant_requests:A list of dictionary containing the grant request parameters.
+    """
+    assert len(grant_requests) == len(grant_ids)
+    self.cbsd_id = cbsd_id
+    self.grant_objects = {}
+    for grant_id, grant_request in zip(grant_ids, grant_requests):
+      self.grant_objects[grant_id] = Grant(grant_id, grant_request)
+    self.registration_request = registration_request
+
+  def getCbsdId(self):
+    return self.cbsd_id
+
+  def getGrantObject(self, grant_id):
+      return self.grant_objects[grant_id]
+
+  def getRegistrationRequest(self):
+    return self.registration_request
+
+  def hasActiveGrant(self):
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isActive():
+        return True
+    return False
+
+  def hasAuthorizedGrant(self):
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isGrantAuthorizedInLastHeartbeat():
+        return True
+    return False
+
+  def constructHeartbeatRequestForAllActiveGrants(self):
+    """Construct list of heartbeat requests of all active grants."""
+    heartbeat_requests = []
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isActive():
+        heartbeat_requests.append(grant_object.constructHeartbeatRequest())
+    return heartbeat_requests
+
+  def getOperationParamsOfAllAuthorizedGrants(self):
+    """Returns the list of operation params of all authorized grants."""
+    operation_params = []
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isGrantAuthorizedInLastHeartbeat():
+        operation_params.append(grant_object.getRequestOperationParam())
+    return operation_params
+
+
+class DomainProxy(object):
+  """Performs Domain Proxy related operations.
+
+  This class stores Domain Proxy related information like the SSL certificate, key to be used and the set of
+  CBSD objects belonging to this Domain Proxy. Bulk operations like registration, grant and heartbeat
+  procedures are performed on all of the CBSDs belonging to this Domain Proxy.
+  """
+  # Define an enumeration class named ResponseCodes with members
+  # 'SUCCESS'
+  class RelinquishmentResponseCodes(Enum):
+    SUCCESS = 0
```

Comment:
ResponseCodes already exists, no need to create a new enum just for relinquishment. Move ResponseCodes out of the Grant class and use that instead, arguably it should be in a separate file.

Reply:
Agreed we added response code in different file and used the same.